messages/compiler/cpp/serialize.cpp

```diff
@@ -69,8 +74,14 @@ void deserialize(uint8_t*& start, const uint8_t* end, T& t) {
       throw NoDataLeftError();
     }
     t = 0;
-    for (auto i = 0u; i < sizeof(T); i++) {
-      t |= ((*(start + i)) << (i * 8));
+    if constexpr (kByteOrder == ByteOrder::kBigEndian) {
+      for (auto i = 0u; i < sizeof(T); i++) {
+        t |= (static_cast<std::make_unsigned_t<T>>((*(start + sizeof(T) - 1 - i))) << (i * 8));
+      }
```

Comment:
From what I understand the previous version didn't work, since a byte was bit shifted?

Reply:
Yes, instead of type T.